extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/export/rya/BindingSetSerializer.java

```diff
@@ -0,0 +1,137 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.rya.indexing.pcj.fluo.app.export.rya;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.util.ArrayList;
+import java.util.Map;
+
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.log4j.Logger;
+import org.apache.rya.api.domain.RyaType;
+import org.apache.rya.api.resolver.RdfToRyaConversions;
+import org.apache.rya.indexing.pcj.storage.accumulo.VisibilityBindingSet;
+import org.openrdf.model.URI;
+import org.openrdf.model.Value;
+import org.openrdf.model.ValueFactory;
+import org.openrdf.model.impl.URIImpl;
+import org.openrdf.model.impl.ValueFactoryImpl;
+import org.openrdf.model.vocabulary.XMLSchema;
+import org.openrdf.query.Binding;
+import org.openrdf.query.BindingSet;
+import org.openrdf.query.impl.ListBindingSet;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+
+public class BindingSetSerializer implements Serializer<VisibilityBindingSet>, Deserializer<VisibilityBindingSet> {
+    private static final ThreadLocal<Kryo> kryos = new ThreadLocal<Kryo>() {
+        @Override
+        protected Kryo initialValue() {
+            Kryo kryo = new Kryo();
+            return kryo;
+        };
+    };
+
+    @Override
+    public VisibilityBindingSet deserialize(String topic, byte[] data) {
+        KryoInternalSerializer internalSerializer = new KryoInternalSerializer();
+        Input input = new Input(new ByteArrayInputStream(data));
+        return internalSerializer.read(kryos.get(), input, VisibilityBindingSet.class);
+        // this is an alternative, or perhaps replace it:
+        // return (new VisibilityBindingSetStringConverter()).convert(new String(data, StandardCharsets.UTF_8), null);
+    }
+
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+        // Do nothing.
+    }
+
+    @Override
+    public byte[] serialize(String topic, VisibilityBindingSet data) {
+        KryoInternalSerializer internalSerializer = new KryoInternalSerializer();
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        Output output = new Output(baos);
+        internalSerializer.write(kryos.get(), output, data);
+        output.flush();
+        byte[] array = baos.toByteArray();
+        return array;
+        // this is an alternative, or perhaps replace it:
+        // return (new VisibilityBindingSetStringConverter()).convert(data, null).getBytes(StandardCharsets.UTF_8);
+    }
+
+    @Override
+    public void close() {
+        // Do nothing.
+    }
+
+    private static Value makeValue(final String valueString, final URI typeURI) {
```

Comment:
This method would no longer be needed if you don't do the RyaType conversion for values.